src/modules/auth/auth.service.ts

```diff
@@ -103,10 +104,26 @@ export default class AuthenticationService {
       const isMatch = await bcrypt.compare(password, user.password);
 
       if (!isMatch) {
+        user.attempts_left -= 1;
+
+        if (user.attempts_left <= 0) {
+          const oneHourAgo = new Date(Date.now() + 60 * 60 * 1000);
+          if (user.time_left && new Date(user.time_left) > oneHourAgo) {
+            throw new CustomHttpException({ message: USER_BANNED, error: 'Forbidden' }, HttpStatus.FORBIDDEN);
+          }
+
+          user.time_left = new Date();
+          user.attempts_left = 3;
+        }
+
+        await this.userService.updateUserAttempts(user.id, user.attempts_left, user.time_left);
```

Comment:
Wrap your db calls in a try catch block to check for easydebugging

Reply:
It is inside a try and catch.